impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java

```diff
@@ -220,23 +228,27 @@ public Stream<ConfigHolder> getBrokerConfigs(Supplier<KafkaEndpoints> endPointCo
             }
 
             if (securityProtocol != null && securityProtocol.contains("SSL")) {
-                if (keytoolCertificateGenerator == null) {
-                    throw new RuntimeException("keytoolCertificateGenerator needs to be initialized when calling KafkaClusterConfig");
+                if (brokerKeytoolCertificateGenerator == null) {
+                    throw new RuntimeException("brokerKeytoolCertificateGenerator needs to be initialized when calling KafkaClusterConfig");
+                }
+                if (clientKeytoolCertificateGenerator == null) {
+                    throw new RuntimeException("clientKeytoolCertificateGenerator needs to be initialized when calling KafkaClusterConfig");
                 }
                 try {
-                    keytoolCertificateGenerator.generateSelfSignedCertificateEntry("test@redhat.com", clientEndpoint.getConnect().getHost(), "KI", "RedHat", null, null,
+                    brokerKeytoolCertificateGenerator.generateSelfSignedCertificateEntry("test@redhat.com", clientEndpoint.getConnect().getHost(), "KI", "RedHat", null, null,
                             "US");
+                    brokerKeytoolCertificateGenerator.generateTrustStore(clientKeytoolCertificateGenerator.getCaCertFilePath(), clientEndpoint.getConnect().getHost());
                 }
                 catch (GeneralSecurityException | IOException e) {
                     throw new RuntimeException(e);
                 }
                 server.put(BrokerSecurityConfigs.SSL_CLIENT_AUTH_CONFIG, "required");
 
-                server.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, keytoolCertificateGenerator.getCertLocation());
-                server.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, keytoolCertificateGenerator.getPassword());
-                server.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, keytoolCertificateGenerator.getCertLocation());
-                server.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, keytoolCertificateGenerator.getPassword());
-                server.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, keytoolCertificateGenerator.getPassword());
+                server.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, brokerKeytoolCertificateGenerator.getTrustStoreLocation());
```

Comment:
THe broker only needs a trust store when SSL client auth is in use.

Reply:
done